hyasinsight_connector.py

```diff
@@ -0,0 +1,1262 @@
+                                                  indicator_value)
+        self.save_progress("Checking the Indicator value")
+        if validate_indicator and validate_indicator == "md5":
+            payload = json.dumps({
+                'applied_filters': {
+                    validate_indicator: indicator_value
+                }
+            })
+            ret_val, response = self._make_rest_call(
+                endpoint,
+                action_result,
+                data=payload,
+                headers=self._headers,
+            )
+
+            if phantom.is_fail(ret_val):
+                # the call to the 3rd party device or service failed, action
+                # result should contain all the error details
+                # for now the return is commented out, but after
+                # implementation, return from here
+                # return action_result.get_status()
+                return ret_val
+
+            # Now post process the data,  uncomment code as you deem fit
+
+            # Add the response into the data section
+            try:
+                all_response[endpoint] = self.get_flatten_json_response(
+                    response)
+
+                action_result.add_data(all_response)
+                return action_result.set_status(phantom.APP_SUCCESS)
+            except:
+                return action_result.set_status(
+                    phantom.APP_ERROR,
+                    "unable to flatten json response.",
+                    None,
+                )
+
+            # Add a dictionary that is made up of the most important values
+            # from data into the summary
+            # summary = action_result.update_summary({})
+            # summary['num_data'] = len(action_result['data'])
+
+            # Return success, no need to set the message, only the status
+            # BaseConnector will create a textual message based off of the
+            # summary dictionary
+            # return action_result.set_status(phantom.APP_SUCCESS)
+
+        return action_result.set_status(
+            phantom.APP_ERROR, MALWARE_RECORD_MD5
+        )
+
+    def _handle_lookup_command_and_control_hash(self, param):
+        self.save_progress(
+            "In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        indicator_value = param['hash']
+        indicator_type = "sha256"
+        # Add an action result object to self (BaseConnector) to represent
+        # the action for this param
+        endpoint = C2ATTRIBUTION
+        all_response = {}
+
+        validate_indicator = self.validating_hash(action_result,
+                                                  indicator_type,
+                                                  indicator_value)
+        self.save_progress("Checking the Indicator value")
+        if validate_indicator and validate_indicator == "sha256":
```

Comment:
I think you may have intended to use `indicator_type` in this comparison instead of the literal.